### PR TITLE
Feature/sup 73 waybill page

### PR DESCRIPTION
**Pull request by IrynaZazyba**
added entities for route creation, implemented:
- display of all invoices,
- modal window for creating a waybill: provides the ability to display the optimal route on the map, the ability change the route by a logistics specialist;

2020-12-08_10-40-47_v1

### CODE REVIEW

File: frontend/src/pages/waybill/Waybill.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    };
+
+    const closeModalAdd = (e) => {
+        setOpenAddModal(e);
+        getWaybill(`/customers/${customerId}/waybills?size=${page.countPerPage}`);
+    };
+
+
+    function parseDestinationLocationsCities(waybill) {
+        let map = waybill.applications.map(a =>
```
Comment (eugenmolchanov) [inline frontend/src/pages/waybill/Waybill.js]: rename map to cities for example

File: frontend/src/pages/waybill/Waybill.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                {parseDestinationLocationsCities(waybill)}
+            </td>
+            <td className="table-text-center">{waybill.lastUpdated}</td>
+            <td className="table-text-center">{waybill.updatedByUsersDto.surname}</td>
+            <td className="table-text-center" style={{width: '45px'}}>
+                <Badge className="badge-status">
+                    {waybill.waybillStatus.replace('_', ' ').toLowerCase()}
+                </Badge></td>
+            <td><FaEdit style={{textAlign: 'center', color: '#1A7FA8'}}
+                        onClick={() => {
```
Comment (eugenmolchanov) [inline frontend/src/pages/waybill/Waybill.js:105]: Why is this anonymous function empty?

File: frontend/src/pages/waybill/Waybill.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                <tbody>
+                {tableRows}
+                </tbody>
+            </Table>
+            <Page page={page} onChange={changePage}/>
+        </React.Fragment>;
+
+    const modals =
+        <React.Fragment>
+            <AddWaybillModal modal={openAddModal} onChange={closeModalAdd}/>
```
Comment (eugenmolchanov) [inline frontend/src/pages/waybill/Waybill.js:139]: Do you pass state as a prop?

File: src/main/java/by/itech/lab/supplier/service/impl/CalculationServiceImpl.java [hunk 4]
```diff
@@ -70,4 +76,48 @@ private void setItemPrice(final ApplicationItem appItem,
         appItem.setCost(new BigDecimal(price));
     }
 
+    @Override
+    public RouteDto calculateOptimalRoute(final List<WarehouseDto> warehouseDto, final WarehouseDto source) {
+        final Deque<WarehouseDto> orderedWarehouses = new ArrayDeque<>();
```
Comment (eugenmolchanov) [inline src/main/java/by/itech/lab/supplier/service/impl/CalculationServiceImpl.java:81]: Why do we need to use Deque here?

**Review by eugenmolchanov — APPROVED** [review]

**Review by alexeyoverchenko — APPROVED** [review]

**Pull request merged**